elastic/kibana
# [Fleet] Fix double policy header layout

## Summary

Fix https://github.com/elastic/kibana/issues/102939. Looks like this is a regression from a big refactor: https://github.com/elastic/kibana/pull/101828

Current implementation adds an explicit check at the `<App />` level for whether we are loading the "Add integration" view and replaces the default header layout (see screenshots).

## How to test

* See the linked issues for details on how to reproduce. This requires starting Elastic+Kibana on basic and setting up an agent for data collection.

* Also check that existing fleet pages still all look correct while loading. This will require going to the different tabs and pressing refresh while looking for any visual regressions.

## To reviewers

* Is `useRouteMatch` the correct thing to use here? My testing and understanding of the docs seem to indicate "yes" 🤷🏻‍♂️ 

## Screenshots

### Loading "Add integrations"

Notice that tabs from the default layout have been removed for this special case.

<img width="1729" alt="Screenshot 2021-06-23 at 13 40 31" src="https://user-images.githubusercontent.com/8155004/123091793-0d319400-d42a-11eb-9368-8a230c4b5a32.png">

### "Add integrations" view after these changes are applied

<img width="1894" alt="Screenshot 2021-06-23 at 13 43 28" src="https://user-images.githubusercontent.com/8155004/123091901-2df9e980-d42a-11eb-8eb1-9f2b24f688c3.png">
